### DISCUSSION TITLE
Growatt SPF ikona   days of production

### DISCUSSION THREAD

**Discussion by LubkoFV** [slipx06/sunsynk-power-flow-card#254] [Q&A]
good day

please add Growatt SPF icon

and my daily production is not working

otherwise top application

Thank you very much

detail:

type: custom:sunsynk-power-flow-card
cardstyle: compact
panel_mode: false
show_solar: true
battery:
  energy: 24000
  shutdown_soc: 100
  show_daily: true
  max_power: 1500
  auto_scale: true
  invert_power: false
  show_absolute: false
  hide_soc: true
  show_remaining_energy: false
solar:
  show_daily: true
  mppts: 2
  max_power: 2000
  auto_scale: true
  dynamic_colour: false
  display_mode: 3
load:
  show_daily: true
  model: Growatt
  max_power: 1500
  auto_scale: true
  dynamic_colour: true
  dynamic_icon: true
  show_aux: false
  invert_aux: false
  additional_loads: 2
  load1_icon: mdi:bathtub
  load1_name: TUV
  essential_name: tuv setap
  load2_icon: mdi:pool
  load2_name: Bazen
grid:
  show_daily_buy: true
  show_daily_sell: false
  show_nonessential: true
  max_power: 1500
  auto_scale: true
  no_grid_colour:
    - 0
    - 138
    - 135
  invert_grid: true
  colour:
    - 206
    - 255
    - 31
entities:
  inverter_voltage_154: sensor.growatt_spf_x_2_inverter_1_pv_voltage_1
  inverter_power_175: sensor.growatt_spf_x_2_load_power
  day_battery_charge_70: sensor.growatt_spf_x_2_battery_energy_in
  day_battery_discharge_71: sensor.growatt_spf_x_2_battery_energy_out
  battery_voltage_183: sensor.growatt_spf_x_2_inverter_1_battery_voltage
  battery_soc_184: sensor.growatt_spf_x_2_battery_state_of_charge
  battery_power_190: sensor.growatt_spf_x_2_battery_power
  battery_current_191: sensor.batteri_soucet_curent
  grid_power_169: '0'
  day_grid_import_76: sensor.growatt_spf_x_2_grid_energy_in
  grid_ct_power_172: sensor.growatt_spf_x_2_grid_power
  day_load_energy_84: sensor.growatt_spf_x_2_load_energy
  nonessential_power: none
  day_pv_energy_108: sensor.oo
  pv1_power_186: sensor.growatt_spf_x_2_inverter_2_pv_power
  pv2_power_187: sensor.growatt_spf_x_2_inverter_1_pv_power
  pv1_voltage_109: sensor.growatt_spf_x_2_inverter_2_pv_voltage_1
  pv1_current_110: sensor.growatt_spf_x_2_inverter_2_pv_current_1
  pv2_voltage_111: sensor.growatt_spf_x_2_inverter_1_pv_voltage_1
  pv2_current_112: sensor.growatt_spf_x_2_inverter_1_pv_current_1
  essential_load1: sensor.tuv_drazice_current_power
  day_grid_export_77: sensor.growatt_spf_x_2_grid_energy_out
  battery_rated_capacity: sensor.growatt_spf_x_2_battery_state_of_charge
  battery_current_direction: sensor.growatt_spf_x_2_inverter_1_battery_current
  essential_load2: sensor.filtrace_bazen_current_power
  inverter_current_164: sensor.pv_curent_1_2
  total_pv_generation: sensor.growatt_spf_x_2_pv_energy
inverter:
  autarky: power
  model: ''
  modern: true
  colour:
    - 255
    - 255
    - 255
  auto_scale: true
  three_phase: false
show_battery: true
show_grid: true
large_font: true
decimal_places: 2

**Comment by slipx06** [comment]
Do you have the correct energy sensors for Daily Production? You need sensors for your Battery, Solar, Grid and load. If your Inverter integration does not provide these you can create your own sensor using the HA  helpers.

**Reply by LubkoFV** [reply]
create your own ???
Could you give me a little hint please?

how to do it

Thank you

**Reply by slipx06** [reply]
Have a look here

https://www.home-assistant.io/integrations/integration/

**Comment by LubkoFV** [comment]
Thank you very much for the advice, I have already tried this process and it worked

the only thing is that if I set it to the day of production, it can't wind me every day

it keeps charging me the next day and not just the current day

**Comment by LubkoFV** [comment]
day 1 10Kwh day 2 10Kwh shows the sum

I would like Kwh from 0 every day

**Reply by slipx06** [reply]
When you create your utility meter make sure that the reset cycle is set to daily

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/7227275/ec196c46-8395-45d0-9aeb-43b3206029df)

**Reply by Roving-Ronin** [reply]
@LubkoFV  Instead of utility_meter have a look at https://github.com/zeronounours/HA-custom-component-energy-meter this expands upon the utility meter, but allows you to call a sensor you've defined for the cost/compensation of the power (import or export) and then have it create a sensor that tracks the cost/compensation. Setup multiple meters with daily / weekly / monthly / yearly as the cycle and you end up with sensors showing the daily / weekly / monthly / yearly of electricity kWh consumed from the grid and its cost, and also the same but for what you exported kWh to the grid and how much you got paid.

**Comment by LubkoFV** [comment]
and if I may ask, would it be possible to add the Growatt giant model

**Reply by slipx06** [reply]
Yes can you share a link to the image of the inverter (png)

**Comment by LubkoFV** [comment]
good day

Thank you very much, you are amazing, you helped me a lot

if there was a picture of the Inventor Growatt spf 5000es, that would be great

but now you are great for me, thank you very much

Thank you

**Comment by LubkoFV** [comment]
thank you very much, you're great

**Comment by LubkoFV** [comment]
otherwise your app is great

and I'm a big fan of you and I'm looking forward to more news

**Comment by LubkoFV** [comment]
I am currently looking at Charge battery automation
Pylontech

**Comment by LubkoFV** [comment]
Hello good day




I can't assemble




I don't know where to write the sensors






otherwise thank you very much

I'm weak at it





-- 
Luboš Lorinc
***@***.***

www.luboslorinc.cz

mob: 604 103 942





---------- Původní e-mail ----------
Od: Roving Ronin ***@***.***>
Komu: slipx06/sunsynk-power-flow-card ***@***.***
github.com>
Kopie: LubkoFV ***@***.***>, Mention ***@***.***>
Datum: 26. 2. 2024 4:41:27
Předmět: Re: [slipx06/sunsynk-power-flow-card] Growatt SPF ikona days of 
production (Discussion #254)
"


@LubkoFV(https://github.com/LubkoFV) Instead of utility_meter have a look at
https://github.com/zeronounours/HA-custom-component-energy-meter
(https://github.com/zeronounours/HA-custom-component-energy-meter) this 
expands upon the utility meter, but allows you to call a sensor you've 
defined for the cost/compensation of the power (import or export) and then 
have it create a sensor that tracks the cost/compensation. Setup multiple 
meters with daily / weekly / monthly / yearly as the cycle and you end up 
with sensors showing the daily / weekly / monthly / yearly of electricity 
kWh consumed from the grid and its cost, and also the same but for what you 
exported kWh to the grid and how much you got paid.

—
Reply to this email directly, view it on GitHub
(https://github.com/slipx06/sunsynk-power-flow-card/discussions/254#discussioncomment-8587336)
, or unsubscribe
(https://github.com/notifications/unsubscribe-auth/A56RM2GB4HHILYVIEYQ4AWTYVP76FAVCNFSM6AAAAABDNCR7ZKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DKOBXGMZTM)
.
You are receiving this because you were mentioned. Message ID: <slipx06/
***@***.***>
"

**Reply by Roving-Ronin** [reply]
@LubkoFV sorry 'assemble' ?  Energy Meter is added to you HA instance, using HACS. Just search for it in the 'Integrations' section, click on it to open it and then finally click on the 'Download' button at the bottom right of screen to install.

How are you charged for your electricity in your country? If you only pay a flat $ per kWh imported, and a flat amount you receive for each kWh exported to the grid then have a look at: 

https://github.com/Roving-Ronin/sunsynk-power-flow-card/blob/master/docs/examples/huawei_packages/huawei_solar_electricity_costs_AU_Red_Energy_TOU.yaml

This creates the energy meters to track the daily / weekly / monthly / yearly kWh exported (and auto creates another sensor for each using the Electricity - Price and Electricity - FIT (feed in tariff...i.e. what you get paid to export) that tracks the $ for each time period.

If you have something more complex, this is my own homes costs where I pay per kWh depending upon that tariff period of the day it was used (peak / shoulder / off peak), so this adds the energy meters, but apart from each day / week / month / year it also creates addition ones within each of those days -->years for each of the tariff periods (i.e. there's a day-peak / day-shoulder / day-offpeak). So whilst I can see how much I used / paid in each of those periods it then uses 'HA Groups' to sum up those periods within a day and the group then gives me the total kWh / Cost $ per day/week/month/year.

As there's now time periods, it also means HA needs to keep checking the time and have an automation that checks the time and then updates the SELECTOR in HA to update what is the current billing period tariff to be counting toward (i.e. peak / shoulder / off peak). Given the data from Huawei inverters is polled every 30 seconds, the automation is then configured to update the selector at 30 seconds after the tariff plans change with the energy retailer  (i.e. 7pm change...so it updates at 7:00:30 pm)

https://github.com/Roving-Ronin/sunsynk-power-flow-card/blob/master/docs/examples/huawei_packages/huawei_solar_electricity_costs_AU_Red_Energy_TOU.yaml

**Comment by LubkoFV** [comment]
I'm so sorry

I'm not very good at it
I'm new to it
I did some tracking, but I'm not very good at it

I would like more statistics and automation for pylontech batteries, but it's very slow and slow


￼￼￼￼
￼



Thank you very much for your help, you are amazing and I thank you very much


I learn it myself from the video or through google or from you


and for that I thank you many times

and thanks for any help





￼

www.luboslorinc.cz
Contact 604103942

**Reply by Roving-Ronin** [reply]
@LubkoFV I have found a few bugs and an error, and also tried to simplify the tasks to setup the card with Huawei. I have submitted a PULL request to update the updated files into Sunsynk Documentation this will appear at:

https://slipx06.github.io/sunsynk-power-flow-card/examples/huawei.html

When you see a table (its rough, I am trying to work out writing in ReST), that is the updated version, please read and re-download the updated configuration and yaml files again, delete the ones you have and reload with the new ones.

**Comment by LubkoFV** [comment]
good day

could I ask for a graphic for the Growatt spf 5000 jpg

as an inventor


Thank you


￼

www.luboslorinc.cz
Contact 604103942



> 27. 2. 2024 v 13:14, Roving Ronin ***@***.***>:
> 
> @LubkoFV <https://github.com/LubkoFV> I have found a few bugs and an error, and also tried to simplify the tasks to setup the card with Huawei. I have submitted a PULL request to update the updated files into Sunsynk Documentation this will appear at:
> 
> https://slipx06.github.io/sunsynk-power-flow-card/examples/huawei.html
> 
> When you see a table (its rough, I am trying to work out writing in ReST), that is the updated version, please read and re-download the updated configuration and yaml files again, delete the ones you have and reload with the new ones.
>

**Comment by slipx06** [comment]
@LubkoFV added to [4.17.0](https://github.com/slipx06/sunsynk-power-flow-card/releases/tag/v4.17.0)

**Comment by LubkoFV** [comment]
￼￼


good day

I couldn't find the picture

Thank you





￼

www.luboslorinc.cz
Contact 604103942

**Reply by slipx06** [reply]
![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/7227275/ef183de1-d3cc-48f8-9f5d-c7fa7428c31e)

**Comment by LubkoFV** [comment]
good day

I understand how it's set up, but there's an error somewhere, so I don't want to display growatt

but it works fine on the app on my mobile

so I'll think about where I'm going wrong

Thank you


na PC
￼

na mobilu

￼



￼

www.luboslorinc.cz
Contact 604103942

**Comment by LubkoFV** [comment]
good day

everything is ok now, it works for me with growatem

Thank you very much

Thank you



￼

www.luboslorinc.cz
Contact 604103942

**Comment by LubkoFV** [comment]
good day




it would not be possible to render graphics from two independent 
inventories, I have two inventories in one phase




so that every inventor has inputs, outputs and sums, shouldn't it be 
granted??




I just have 2X inventor PV, grid, Load is always special

at the end it is cleaned so that it can be seen how everyone is driving, 
what load and vikon




2x inventor in one phase




Thank you




for info

they have already perfected it

it would be more













-- 
Luboš Lorinc
***@***.***

www.luboslorinc.cz

mob: 604 103 942





---------- Původní e-mail ----------
Od: slipx06 ***@***.***>
Komu: slipx06/sunsynk-power-flow-card ***@***.***
github.com>
Kopie: LubkoFV ***@***.***>, Mention ***@***.***>
Datum: 29. 2. 2024 21:46:41
Předmět: Re: [slipx06/sunsynk-power-flow-card] Growatt SPF ikona days of 
production (Discussion #254)
"


image.png (view on web)
(https://github.com/slipx06/sunsynk-power-flow-card/assets/7227275/ef183de1-d3cc-48f8-9f5d-c7fa7428c31e)

—
Reply to this email directly, view it on GitHub
(https://github.com/slipx06/sunsynk-power-flow-card/discussions/254#discussioncomment-8635596)
, or unsubscribe
(https://github.com/notifications/unsubscribe-auth/A56RM2CIJMKRUEJTNGGOQ2DYV6JS3AVCNFSM6AAAAABDNCR7ZKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DMMZVGU4TM)
.
You are receiving this because you were mentioned. Message ID: <slipx06/
***@***.***>
"

**Comment by LubkoFV** [comment]
good morning, if I offended you in any way, I'm very sorry, I'm sorry, have a nice dayLubkoPHOTOmobil:60410394229. 2. 2024 v 21:46, slipx06 ***@***.***>:﻿
image.png (view on web)
